File: sources/242.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    len_s = s.length;
    len_t = t.length;
    if( len_s != len_t ){
        return false;
    }
    if( len_s <= 1 ){
        return s == t;
    } else {
        var dict_s = new Array();
        var dict_t = new Array();
        for(var i=0; i<len_s; i++){
            if(s[i] in dict_s){
                dict_s[s[i]] += 1;
            } else {
                dict_s[s[i]] = 1;
            }
            if(t[i] in dict_t){
                dict_t[t[i]] += 1;
            } else {
                dict_t[t[i]] = 1;
            }
        }
        for(var k in dict_s){
            if(dict_s[k] != dict_t[k]){
                return false;
            }
        }
        return true;
    }
};

File: sources/py/242.py
class Solution:
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        # 一开始没搞清楚题意，模棱两可的题目描述真可怕
        # 题意是说判断给定的两个字符串是否包含相同的字母并且每个字母出现的次数都一样（只是位置不同）
        len_s, len_t = len(s), len(t)
        if len_s != len_t: # 长度不同直接返回False
            return False
        if len_s <= 1:
            return s == t
        set_s, set_t = set(s), set(t)
        if len(set_s) != len(set_t) or set_s != set_t:
            return False
        dict_s = {}
        for c in set_s:
            dict_s[c] = s.count(c)
        dict_t = {}
        for c in set_t:
            dict_t[c] = t.count(c)
        return dict_s == dict_t # 直接判断两个字典是否相同，即是否有相同的字母和对应次数相同
